lxd/patches.go

```diff
@@ -1174,4 +1175,141 @@ func patchStorageZfsUnsetInvalidBlockSettingsV2(_ string, d *Daemon) error {
 	return nil
 }
 
+// patchStorageUnsetInvalidBlockSettings removes invalid block settings from LVM and Ceph RBD volumes.
+func patchStorageUnsetInvalidBlockSettings(_ string, d *Daemon) error {
+	s := d.State()
+
+	// Get all storage pool names.
+	pools, err := s.DB.Cluster.GetStoragePoolNames()
+	if err != nil {
+		// Skip the rest of the patch if no storage pools were found.
+		if api.StatusErrorCheck(err, http.StatusNotFound) {
+			return nil
+		}
+
+		return fmt.Errorf("Failed getting storage pool names: %w", err)
+	}
+
+	volTypeCustom := db.StoragePoolVolumeTypeCustom
+	volTypeVM := db.StoragePoolVolumeTypeVM
+
+	poolIDNameMap := make(map[int64]string, 0)
+	poolVolumes := make(map[int64][]*db.StorageVolume, 0)
+
+	err = s.DB.Cluster.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
+		for _, pool := range pools {
+			// Get storage pool ID.
+			poolID, err := tx.GetStoragePoolID(ctx, pool)
+			if err != nil {
+				return fmt.Errorf("Failed getting storage pool ID of pool %q: %w", pool, err)
+			}
+
+			driverName, err := tx.GetStoragePoolDriver(ctx, poolID)
+			if err != nil {
+				return fmt.Errorf("Failed getting storage pool driver of pool %q: %w", pool, err)
+			}
+
+			if !shared.ValueInSlice[string](driverName, []string{"lvm", "ceph"}) {
```

Comment:
@roosterfish do we need a fix for dir or btrfs?

Reply:
They already block creation of such volumes:

```bash
ulian@thinkpad:~$ lxc storage create btrfs btrfs
Storage pool btrfs created
julian@thinkpad:~$ lxc storage volume create btrfs vol --type block block.filesystem=xfs
Error: Invalid option for volume "default_vol" option "block.filesystem"
julian@thinkpad:~$ lxc storage create dir dir
Storage pool dir created
julian@thinkpad:~$ lxc storage volume create dir vol --type block block.filesystem=xfs
Error: Invalid option for volume "default_vol" option "block.filesystem"

```